src/parser/jobs/rpr/modules/ImmortalSacrifice.tsx

```diff
@@ -0,0 +1,147 @@
+import {t} from '@lingui/macro'
+import {Trans} from '@lingui/react'
+import React from 'react'
+import {DataLink} from '../../../../components/ui/DbLink'
+import {Event, Events} from '../../../../event'
+import {Analyser} from '../../../core/Analyser'
+import {filter} from '../../../core/filter'
+import {dependency} from '../../../core/Injectable'
+import Checklist, {Requirement, Rule} from '../../../core/modules/Checklist'
+import {Data} from '../../../core/modules/Data'
+import {DISPLAY_ORDER} from './DISPLAY_ORDER'
+
+interface ImmortalSacrificeWindow {
+	startTime: number,
+	endTime?: number,
+	/**
+	 * If this window was closed, by consuming the stacks, or it expired
+	 */
+	closed: boolean,
+	stackCount: number,
+	consumed: boolean,
+}
+
+export default class ImmortalSacrifice extends Analyser {
+
+	static override handle = 'immortalsacrifice'
+	static override title = t('rpr.immortal-sacrifice.title')`Immortal Sacrifice stacks consumed`
+	static override debug = false
+
+	@dependency private data!: Data
+	@dependency private checklist!: Checklist
+
+	private windows: ImmortalSacrificeWindow[] = []
+
+	override initialise() {
+		super.initialise()
+
+		const playerFilter = filter<Event>().source(this.parser.actor.id)
+
+		this.addEventHook(playerFilter.type('statusApply').status(this.data.statuses.IMMORTAL_SACRIFICE.id), this.onImmortalSacrificeStack)
+		this.addEventHook(playerFilter.type('statusRemove').status(this.data.statuses.IMMORTAL_SACRIFICE.id), this.onImmortalSacrificeExpiration)
+		this.addEventHook(playerFilter.type('action').action(this.data.actions.PLENTIFUL_HARVEST.id), this.onPlentifulHarvest)
+
+		this.addEventHook('complete', this.onComplete)
+	}
+
+	private onImmortalSacrificeStack(event: Events['statusApply']) {
+		let window = this.getOpenWindow()
+
+		if (!window) {
+			window = {
+				startTime: event.timestamp,
+				closed: false,
+				consumed: false,
+				stackCount: 0,
+			}
+
+			this.windows.push(window)
+		}
+
+		window.stackCount = event.data || window.stackCount + 1
+	}
+
+	private onPlentifulHarvest(event: Events['action']) {
+		const window = this.getOpenWindow()
+
+		if (!window) {
+			this.debug(`Plentiful Harvest executed but there were no immortal sacrifice window open: ${this.windows}`)
```

Comment:
It's probably worth triggering Broken Log for this, so that we get notifications if it's happening (and so that the log displays a warning since other things might be mis-handled due to the bad data).

You can find an example in the `onNastrondCast` function in DRG's BloodOfTheDragon module.

Reply:
Thanks, added.